### ISSUE TITLE
Unable to run emacs batch mode

### ISSUE THREAD

**Issue by shivams** [doomemacs/doomemacs#6494]
### What did you expect to happen?

I am trying to run Emacs [in batch mode (in terminal) for exporting org-mode files to PDF](https://emacs.stackexchange.com/questions/10392/export-an-org-mode-file-as-a-pdf-via-command-line), using the following command:

```shell
emacs --debug-init -u "$(id -un)" --batch --eval '(load user-init-file)' notes.org -f org-latex-export-to-pdf
```

Note that the `--batch` option disables loading of any init files by default. Hence I am loading the init files using `eval` in the above command. If I simply use the `--batch` mode, the export runs fine, however it fails to load many org-mode packages and the exported pdf comes out to be incomplete (e.g. without syntax highlighting). Here is that bare-bones command:

```shell
emacs --batch notes.org -f org-latex-export-to-pdf
```

It does reproduce on Vanilla Doom (without my private config).

### What actually happened?

However, it throws the error:

```
Loading /etc/emacs/site-start.d/00debian.el (source)...
Loading /etc/emacs/site-start.d/50autoconf.el (source)...
Loading /etc/emacs/site-start.d/50dictionaries-common.el (source)...
Loading debian-ispell...
Loading /var/cache/dictionaries-common/emacsen-ispell-default.el (source)...
Loading /var/cache/dictionaries-common/emacsen-ispell-dicts.el (source)...
Debugger entered--Lisp error: (void-function doom-initialize-packages)
  (doom-initialize-packages)
  (if noninteractive (doom-initialize-packages) (doom-initialize-core) (doom-initialize-modules))
  eval-buffer(#<buffer  *load*> nil "/home/dean/.emacs.d/init.el" nil t)  ; Reading at buffer position 2724
  load-with-code-conversion("/home/dean/.emacs.d/init.el" "/home/dean/.emacs.d/init.el" t t)
  load("/home/dean/.emacs.d/init" noerror nomessage)
  startup--load-user-init-file(#f(compiled-function () #<bytecode 0x158fc495cfa1>) #f(compiled-function () #<bytecode 0x158fc495cfb5>) t)
  command-line()
  normal-top-level()
```

### Describe your attempts to resolve the issue

I added the `--debug-init` option to trace the error but couldn't figure why it was not able to load the init file.

### Steps to reproduce

1. Have an org-mode file ready for export to pdf (through latex)
2. Open a terminal and run this command to export: `emacs -u "$(id -un)" --batch --eval '(load user-init-file)' notes.org -f org-latex-export-to-pdf`
3. It fails to export throwing the error: `Symbol’s function definition is void: doom-initialize-packages`
4. Note that the export runs if I don't load the Doom emacs init files: `emacs --batch notes.org -f org-latex-export-to-pdf`. The `--batch` command by default disables loading of any init files.

### System Information

https://pastebin.com/XTFC1RSj

**Comment by hlissner** [comment]
**The problem:** `user-init-file` is set to `~/.emacs.d/init.el` (or `~/.config/emacs/init.el`) by default. Doom no longer has an init.el file (since 1402db5129497d5bfb1de3b781f0f778c2aefb0d). So it has nothing to load. Instead, Doom has replaced it with `~/.emacs.d/core/core-start.el` *and* in the future, this file will be replaced with a dynamically generated elisp file, so it's not safe to load explicitly, but it'd look like this: `emacs --batch -l ~/.emacs.d/core/core-start.el notes.org -f org-latex-export-to-pdf`

**The solution:** I'm working on a `doom batch` command for running commands against your Doom config, which will have all the batch options `emacs` supports and more, *but* that doesn't help you now, so here's a more immediate solution: set up a doomscript:

1. Make sure `~/.emacs.d/bin` is in your `$PATH`.
2. Make sure `~/.doom.d/bin` is in your `$PATH`.
3. Put this in `~/.doom.d/bin/export-notes`:
 
   ```sh
   #!/usr/bin/env doomscript

   (defcli! export-notes (&args files)
     (require 'core-start) ; load your user config
     (dolist (file files)
       (find-file file) ; open an org file
       (org-latex-export-to-pdf)))

   (run! "export-notes" (cdr (member "--" argv)))
   ```
4. Make this script executable: `chmod +x export-notes`
5. Now you have a Doom script: `export-notes notes.org`

-----

If you're interested in the capabilities of `defcli!` (it's quite powerful), you'll find [more examples of doomscripts here](https://gist.github.com/hlissner/ba8c3b4c6f37c24ff27b72194942b7aa), and I recommend looking at `defcli!`'s docstring if you want more details (you need to load it `M-x load-library RET doom-cli-lib`, then `C-h f defcli\!`).

Hope that helps!

**Comment by hlissner** [comment]
Have you updated Doom lately? `bin/doomscript` was only added a couple days ago in e5b7edcd8.

**Comment by shivams** [comment]
Yes, updating to the latest version brought doomscript the export-notes command is now working.

**Comment by Ramblurr** [comment]
Should this doomscript feature support manually installed packages (those provded in my own `packages.el` file)?

I've a simple test script like so:

```elisp
#!/usr/bin/env doomscript

(defcli! (publish) ()
  "docstring"

  (require 'ox-publish)
  (require 'ox-gemini))

(run! "publish" (cdr (member "--" argv)))

```

This script fails with:
```
Message: File is missing
  Details: ("Cannot open load file" "No such file or directory" "ox-gemini")
  Backtrace:
    (require ox-gemini)
...
```

My `packages.el` has `(package! ox-gemini)`, and my `config.el` has `(use-package ox-gemini)`.

When using doom normally the ox-gemini functions are available.

**Comment by ronnac** [comment]
There doesn't seem to be a doom batch command by now. How can I run the following batch in doom: 
`emacs -batch -l ~/.emacs -eval '(org-batch-agenda "t")'` ?

**Comment by hlissner** [comment]
Doom's core is in a bit of a weird state at the moment; I'm working some large architectural changes under the hood, so for the time being you'll need to call `doom-initialize`, then `doom-startup` to load your config (no need to load `doom-start`). You'll also need to call `doom-modules-initialize` for standalone doomscripts:

```
#!/usr/bin/bash doomscript 

(defcli! listtodos () 
  (doom-modules-initialize)
  (doom-initialize t) 
  (doom-startup)
  (require 'org) 
  (org-batch-agenda "t")) 

(run! "listtodos")
```

If you wanted, you can also acll `doom-modules-initialize` at top-level, before the `run!` call.

Anyhow, once I've deployed these upcoming changes, this will fail (because `doom-modules-initialize` and `doom-initialize` will no longer exist). When this happens, you'll only need `(doom-startup)` from that point on.
